src/client/gfs_client.h

```diff
@@ -0,0 +1,44 @@
+#ifndef CLIENT_GFS_CLIENT_H
+#define CLIENT_GFS_CLIENT_H
+
+#include "google/protobuf/stubs/status.h"
+#include "google/protobuf/stubs/statusor.h"
+
+namespace gfs {
+namespace client {
+
+// Use the Status and Statusor objects provided by protobuf
+using google::protobuf::util::Status;
+using google::protobuf::util::StatusOr;
+
+/* Data type for the return of read operation. This includes how many bytes have been
+ * read and a pointer of the buffer */
+struct Data {
+   int bytesRead;
```

Comment:
Q: Do we want to support it? it may mean that the client GFS and gRPC has to now communicate this --> if so, sync with matthew on this so he's aware?